plat/kvm/arm/setup.c

```diff
@@ -238,6 +425,17 @@ void _libkvmplat_start(void *dtb_pointer)
 	uk_pr_info("      stack top: %p\n",
 		   (void *) _libkvmplat_cfg.bstack.start);
 
+	_init_dtb_cpu();
+
+	if (cpu0 < 0) {
+		uint64_t mpidr_reg = SYSREG_READ32(mpidr_el1);
```

Comment:
@rene Would it also be useful to initialize cpu0 with bits [11:8] (cpuid) of mpidr? And define a function that extracts this bits for the current cpu, similar to [this one](https://github.com/unikraft/unikraft/pull/244/files#diff-7f41a81bfcd4b955f5dcfb2e2b5434fe21135e09b7f654a7f5da6937652c1c76R198).
I'm thinking that this would keep a difference between physical and logical ids.

Reply:
@razvanvirtan , yes, I think IDs of all cores must be initialized. IMO the function to get the ID should return only the affinity fields grouped together, in case of Cortex-A we have the following format for MIPDR_EL1:
![image](https://user-images.githubusercontent.com/62743/133781863-2abba07d-ad49-4fc1-af54-793c7fdc7c99.png)
So the function should return in a single value Aff? grouped into a single value, like this: |Aff3|Aff2|Aff1|Aff0|

Reply:
@rene I think the physical ids are already initialized for all cores in `_init_dtb_cpu()` (they are read from the device tree and stored in the `cpus[]` vector), so we don't need to use `ukplat_lcpu_id()` for that.
However, `ukplat_lcpu_id()` will provide a way to find the id of the core it is running on. This can't be done by using the cpus vector, since we don't know the index (logical id) of the current cpu. Therefore, finding the boot core physical id will be only one usecase for this function.